modules/system-info/reporters/server.php

```diff
@@ -410,4 +403,63 @@ public function get_elementor_library() {
 			'value' => 'Connected',
 		];
 	}
+
+	/**
+	 * Get permissions by keys. If empty keys, get all permissions.
+	 * Be aware that you can only use key paths that are defined in the get_paths() method.
+	 *
+	 * @param array $key_paths
+	 * @return array {read: bool, write: bool, execute: bool}
+	 */
+	public function get_permissions( array $key_paths = [] ):array {
+		$paths = $this->get_paths();
+		$all_key_paths = array_keys( $paths );
+
+		$key_paths = empty( $key_paths ) ? $all_key_paths : array_intersect( $key_paths, $all_key_paths );
+
+		$permissions = [];
+
+		foreach ( $key_paths as $key_path ) {
+			$path = $paths[ $key_path ];
+
+			$permissions[ $key_path ] = $this->get_path_permissions( $path );
+		}
+
+		return $permissions;
+	}
+
+	/**
+	 * Get all the paths that we might need to check permissions.
+	 *
+	 * @return array
+	 */
+	private function get_paths(): array {
+		return [
+			static::KEY_PATH_WP_ROOT_DIR => ABSPATH,
+			static::KEY_PATH_HTACCESS_FILE => file_exists( ABSPATH . '/.htaccess' ) ? ABSPATH . '/.htaccess' : '',
+			static::KEY_PATH_WP_CONTENT_DIR => WP_CONTENT_DIR,
+			static::KEY_PATH_UPLOADS_DIR => wp_upload_dir()['basedir'] ?? '',
+			static::KEY_PATH_ELEMENTOR_UPLOADS_DIR => ! empty( wp_upload_dir()['basedir'] ) ? wp_upload_dir()['basedir'] . '/elementor' : '',
+		];
+	}
+
+	/**
+	 * @param $path
+	 * @return array{read: bool, write: bool, execute: bool}
+	 */
+	private function get_path_permissions( $path ): array {
```

Comment:
tests?

Reply:
I tried to add tests, but couldn't find a way to change permissions.